src/firefly/js/visualize/ui/ColorDialog.jsx

```diff
@@ -2,248 +2,226 @@
+                        <FieldGroup groupKey={RED_PANEL} keepState={true} >
+                            <ColorBandPanel groupKey={RED_PANEL} band={Band.RED} fields={rFields}
+                                            plot={plot} key={Band.RED.key}/>
+                        </FieldGroup>
+                    </Tab>
+                    }
+
+                    {plotState.isBandUsed(Band.GREEN) &&
+                    <Tab name='Green' id='green'>
+                        <FieldGroup groupKey={GREEN_PANEL} keepState={true} >
+                            <ColorBandPanel groupKey={GREEN_PANEL} band={Band.GREEN} fields={gFields}
+                                            plot={plot}key={Band.GREEN.key}/>
+                        </FieldGroup>
+                    </Tab>
+                    }
+
+                    {plotState.isBandUsed(Band.BLUE) &&
+                    <Tab name='Blue' id='blue'>
+                        <FieldGroup groupKey={BLUE_PANEL} keepState={true} >
+                            <ColorBandPanel groupKey={BLUE_PANEL} band={Band.BLUE} fields={bFields}
+                                            plot={plot} key={Band.BLUE.key}/>
+                        </FieldGroup>
+                    </Tab>
+                    }
+                </FieldGroupTabs>
+                <CompleteButton
+                    groupKey={['colorDialogTabs',RED_PANEL,GREEN_PANEL,BLUE_PANEL]}
+                    closeOnValid={false}
+                    style={{padding: '2px 0 7px 10px'}}
+                    onSuccess={replot}
+                    onFail={invalidMessage}
+                    text='Refresh'
+                    dialogId='ColorStretchDialog'
+                    includeUnmounted={true}
+                />
+            </FieldGroup>
+        </div>
+
+    );
 
-    propTypes: {
-        groupKey : React.PropTypes.string.isRequired,
-        band : React.PropTypes.object.isRequired
-    },
+}
 
-    getInitialState() {
-        return {fields : FieldGroupUtils.getGroupFields(this.props.groupKey)};
-    },
+function renderStandardView(plot,fields) {
+    return (
+        <div>
+            <FieldGroup groupKey={NO_BAND_PANEL} keepState={true} >
+                <ColorBandPanel groupKey={NO_BAND_PANEL} band={Band.NO_BAND} fields={fields} plot={plot}/>
+                <CompleteButton
+                    closeOnValid={false}
+                    style={{padding: '2px 0 7px 10px'}}
+                    onSuccess={replot}
+                    onFail={invalidMessage}
+                    text='Refresh'
+                    dialogId='ColorStretchDialog'
+                />
+            </FieldGroup>
+        </div>
+       );
+}
 
-    onClick() {
-        this.doClose();
-    },
+function replot(request) {
+    if (request.colorDialogTabs) {
+        replot3Color(
+            request.redPanel,request.greenPanel,
+            request.bluePanel,
+            request.colorDialogTabs.colorTabs);
```

Comment:
so the fieldgroup for red, green, and blue are in the parallel position in the request object as request.colorDialogTabs? I thought they represent fieldgroup inside another fieldgroup?